modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1323,11 +1323,13 @@ def from_labels(self) -> "PandasDataframe":
                 if "index" not in self.columns
                 else "level_{}".format(0)
             ]
-        new_dtypes = None
-        if self.has_materialized_dtypes:
-            names = tuple(level_names) if len(level_names) > 1 else level_names[0]
-            new_dtypes = self.index.to_frame(name=names).dtypes
-            new_dtypes = pandas.concat([new_dtypes, self.dtypes])
+        names = tuple(level_names) if len(level_names) > 1 else level_names[0]
+        new_dtypes = self.index.to_frame(name=names).dtypes
+        try:
+            new_dtypes = ModinDtypes.concat([new_dtypes, self._dtypes])
+        except NotImplementedError:
+            # can raise on duplicated labels
```

Comment:
If the old procedure worked in this case, then it can still be used.

Reply:
@dchigarev can I merge it, or will you look at this?

Reply:
I'll look at this

Reply:
added the old logic to `ModinDtypes.concat()`, however, it's still worth keeping this `try-except` block here, since it still can raise a `NotImplementedError`